src/parser/lval.js

```diff
@@ -378,4 +380,18 @@ export default class LValParser extends NodeUtils {
       }
     }
   }
+
+  checkToRestConversion(node: SpreadElement): void {
+    const validArgumentTypes = [
```

Comment:
It's ok to leave the type definition as is, because `RestElement` is also used by array destructuring (and the argument of a rest element inside an array pattern can be a pattern).